roll/testing.py

```diff
@@ -130,6 +132,8 @@ def encode_body(self, body, headers):
     async def request(self, path, method='GET', body=b'', headers=None,
                       content_type=None):
         headers = headers or {}
+        for key, value in self.headers.items():
+            headers.setdefault(key, value)
```

Comment:
Why not `headers.update(self.default_headers)`?

Reply:
So `headers` from the current request context (passed as argument) will have precedence over the default ones.